From ce32478594b4afc26c91ed63eb17c65ca988158c Mon Sep 17 00:00:00 2001
From: 4piu <16668056+4piu@users.noreply.github.com>
Date: Mon, 7 Mar 2022 16:38:08 +0800
Subject: [PATCH] Add support for NEC VersaPro VG-S

The brightness control key (Fn+F7 Fn+F8) and touchpad toggle key (Fn + Space) do not work on the NEC VersaPro VG-S laptop. Add the keycode to fix the problem.

(cherry picked from commit 19a04e807129e7790a0f2950cf67d79fe0df85cb)

Related: #2087778
---
 hwdb.d/60-keyboard.hwdb | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/hwdb.d/60-keyboard.hwdb b/hwdb.d/60-keyboard.hwdb
index 2606d0b66c..73b26089ab 100644
--- a/hwdb.d/60-keyboard.hwdb
+++ b/hwdb.d/60-keyboard.hwdb
@@ -1345,6 +1345,17 @@ evdev:name:MSI Laptop hotkeys:dmi:bvn*:bvr*:bd*:svn*:pnM[iI][cC][rR][oO]-S[tT][a
  KEYBOARD_KEY_0213=f22
  KEYBOARD_KEY_0214=f23
 
+##########################################
+# NEC
+##########################################
+
+# VersaPro VG-S
+evdev:atkbd:dmi:bvn*:bvr*:bd*:svnNEC:pnPC-VK22TGSGS:pvr*
+ KEYBOARD_KEY_a8=f21                                    # Fn+Space touchpad toggle
+ KEYBOARD_KEY_67=brightnessdown                         # Fn+F7 brightness down
+ KEYBOARD_KEY_65=brightnessup                           # Fn+F8 brightness up
+ KEYBOARD_KEY_71=battery                                # Fn+F4 ECO
+
 ###########################################################
 # Olimex
 ###########################################################
